projects/js-packages/components/components/scan-report/index.tsx

```diff
@@ -0,0 +1,208 @@
+			layout: {
+				primaryField: FIELD_STATUS,
+			},
+		},
+		list: {
+			...baseView,
+			fields: [ FIELD_STATUS, FIELD_VERSION ],
+			layout: {
+				primaryField: FIELD_NAME,
+				mediaField: FIELD_ICON,
+			},
+		},
+	};
+
+	/**
+	 * DataView view object - configures how the dataset is visible to the user.
+	 *
+	 * @see https://developer.wordpress.org/block-editor/reference-guides/packages/packages-dataviews/#view-object
+	 */
+	const [ view, setView ] = useState< View >( {
+		type: 'table',
+		...defaultLayouts.table,
+	} );
+
+	/**
+	 * DataView fields - describes the visible items for each record in the dataset.
+	 *
+	 * @see https://developer.wordpress.org/block-editor/reference-guides/packages/packages-dataviews/#fields-object
+	 */
+	const fields = useMemo( () => {
+		const iconHeight = 20;
+
+		const result = [
+			{
+				id: FIELD_STATUS,
+				label: __( 'Status', 'jetpack-components' ),
+				render( { item } ) {
+					if ( item.checked ) {
+						if ( item.threats.length > 0 ) {
+							return (
+								<Tooltip
+									className={ styles.tooltip }
+									text={ __( 'Threat detected.', 'jetpack-components' ) }
+								>
+									<div className={ styles.icon }>
+										<ShieldIcon variant="warning" height={ iconHeight } />
+									</div>
+								</Tooltip>
+							);
+						}
+						return (
+							<Tooltip
+								className={ styles.tooltip }
+								text={ __(
+									'No known threats found that affect this version.',
+									'jetpack-components'
+								) }
+							>
+								<div className={ styles.icon }>
+									<ShieldIcon variant="success" height={ iconHeight } />
+								</div>
+							</Tooltip>
+						);
+					}
+					return (
+						<Tooltip
+							className={ styles.tooltip }
+							text={ __(
+								'This item was added to your site after the most recent scan. We will check for threats during the next scheduled one.',
+								'jetpack-components'
+							) }
+						>
+							<div className={ styles.icon }>
+								<ShieldIcon variant="info" height={ iconHeight } />
+							</div>
+						</Tooltip>
+					);
+				},
+			},
+			{
+				id: FIELD_TYPE,
+				label: __( 'Type', 'jetpack-components' ),
+				elements: TYPES,
+				render( { item } ) {
+					return item.type ? item.type.charAt( 0 ).toUpperCase() + item.type.slice( 1 ) : '';
+				},
+			},
+			{
+				id: FIELD_NAME,
+				label: __( 'Name', 'jetpack-components' ),
+				enableGlobalSearch: true,
+				render( { item } ) {
+					return item.name ? item.name : '';
+				},
+			},
+			{
+				id: FIELD_VERSION,
+				label: __( 'Version', 'jetpack-components' ),
+				enableGlobalSearch: true,
+				render( { item } ) {
+					return item.version ? item.version : '';
+				},
+			},
+			{
+				id: FIELD_ICON,
+				label: __( 'Icon', 'jetpack-components' ),
+				render( { item } ) {
+					return (
+						<div className={ styles.threat__media }>
+							<Icon icon={ ICONS[ item.type ] } />
+						</div>
+					);
+				},
+			},
```

Comment:
```suggestion
			...( view.type === 'list' ? [ {
				id: FIELD_ICON,
				label: __( 'Icon', 'jetpack-components' ),
				enableHiding: false,
				getValue( { item } ) {
					return ICONS[ item.type ] || '';
				},
				render( { item } ) {
					return (
						<div className={ styles.threat__media }>
							<Icon icon={ ICONS[ item.type ] } />
						</div>
					);
				},
			} ] : [] ),
```

- Since there will not be a `threat.icon` value available, we can use `getValue` for this field.
- As it is only relevant in list view, we could also conditionally include it in the result and disable `enableHiding` to keep the field completely hidden in table view.

(This inline suggestion would require an update to the `useMemo` deps this code is wrapped in.)